src/basic/string-util.c

```diff
@@ -583,6 +583,22 @@ static void advance_offsets(
                 shift[1] += size;
 }
 
+char *strip_newline(char **ibuf, size_t *_isz) {
```

Comment:
pretty sure we already have helpers to strip strings of whitespace

Reply:
We do have `truncate_nl()`, but it has 2 major disadvantages: 

1. it uses non-performant strcspn (in most libc implementations it'll build a table of "reject" characters and this will take much more instructions to process)
2. it is unable to return truncated string length, so it requires an additional strlen() call to set `message_len`

From my perspective it's too much of work if we gonna to call this for each message we're going to show. What do you think?

Reply:
I have no opinion w.r.t. performances, I'll leave that to @poettering and @DaanDeMeyer

Reply:
Below we agreed to use `strcspn()`, so implemented in a better way.